sdk/lusid/utilities/api_client_builder.py

```diff
@@ -13,62 +13,46 @@
 
 class ApiClientBuilder:
 
-    def build(self, api_configuration):
+    @staticmethod
+    def build(api_configuration, okta_response_handler = None):
         """
         :param api_configuration: name api configuration file
+        :param okta_response_handler: optional handler for Okta response
         :return: ApiClient correctly configured with credentials and host
         """
 
-        # Load our configuration details from the environment variables
-        token_url = os.getenv("FBN_TOKEN_URL", None)
-        api_url = os.getenv("FBN_LUSID_API_URL", None)
-        username = os.getenv("FBN_USERNAME", None)
-        password_raw = os.getenv("FBN_PASSWORD", None)
-        client_id_raw = os.getenv("FBN_CLIENT_ID", None)
-        client_secret_raw = os.getenv("FBN_CLIENT_SECRET", None)
-        app_name = os.getenv("FBN_APP_NAME", "")
-
-        # If any of the environmental variables are missing use a local secrets file
-        if token_url is None or username is None or password_raw is None or client_id_raw is None \
-                or client_secret_raw is None or api_url is None:
-
-            dir_path = os.path.dirname(os.path.realpath(__file__))
-            with open(os.path.join(dir_path, api_configuration), "r") as secrets:
-                config = json.load(secrets)
-
-            token_url = os.getenv("FBN_TOKEN_URL", config["api"]["tokenUrl"])
-            username = os.getenv("FBN_USERNAME", config["api"]["username"])
-            password = pathname2url(os.getenv("FBN_PASSWORD", config["api"]["password"]))
-            client_id = pathname2url(os.getenv("FBN_CLIENT_ID", config["api"]["clientId"]))
-            client_secret = pathname2url(os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"]))
-            api_url = os.getenv("FBN_LUSID_API_URL", config["api"]["apiUrl"])
-            app_name = os.getenv("FBN_APP_NAME", config["api"].get("applicationName", ""))
-
-        else:
-            password = pathname2url(password_raw)
-            client_id = pathname2url(client_id_raw)
-            client_secret = pathname2url(client_secret_raw)
+        # Load the configuration
+        configuration = ApiConfigurationLoader().load(api_configuration)
 
         # Prepare our authentication request
-        token_request_body = ("grant_type=password&username={0}".format(username) +
-                              "&password={0}&scope=openid client groups".format(password) +
-                              "&client_id={0}&client_secret={1}".format(client_id, client_secret))
+        token_request_body = f"grant_type=password&username={configuration.username}" \
```

Comment:
You're going to persuade me to use f strings aren't you. Hard to see how format is better.

Reply:
👍